pkg/logql/shardmapper_test.go

```diff
@@ -417,6 +421,35 @@ func TestMappingStrings(t *testing.T) {
 	}
 }
 
+// Test that mapping of queries for operation types that have probabilistic
+// sharding options, but whose sharding is turned off, are not sharded on those operations.
+func TestMappingStrings_NoProbabilisticSharding(t *testing.T) {
+	m := NewShardMapper(ConstantShards(2), nilShardMetrics, []string{})
+	for _, tc := range []struct {
+		in  string
+		out string
+	}{
+		{ // This should result in the same downstream sharding of the max and not the inner quantile regardless of whether quantile sharding is turned on.
+			in:  `max by (status)(quantile_over_time(0.70, {a=~".+"} | logfmt | unwrap value [1s]))`,
+			out: `maxby(status)(downstream<maxby(status)(quantile_over_time(0.7,{a=~".+"}|logfmt|unwrapvalue[1s])),shard=0_of_2>++downstream<maxby(status)(quantile_over_time(0.7,{a=~".+"}|logfmt|unwrapvalue[1s])),shard=1_of_2>)`,
```

Comment:
Ok, at first I thought this was wrong but this works because `quantile_over_time` has no by clause right?

Reply:
This query has `max` as the top level aggregation, and AFAIK currently we're choosing to only shard the top level aggregation.